### PR TITLE
[Bug](point query) cancel future when meet timeout in PointQueryExec

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

1. cancel future when meet timeout and add config to modify rpc timeout
2. add config to modify numof BackendServiceProxy since under high concurrent work load GRPC channel will be blocked

…rrent workd load BRPC channel will be blocked frenquently
Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 54.28 seconds
 stream load tsv:          500 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          86.8 seconds inserted 10000000 Rows, about 115K ops/s
 storage size: 17162337660 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230706170314_clickbench_pr_173565.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.24 seconds
 stream load tsv:          441 seconds loaded 74807831229 Bytes, about 161 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          70.0 seconds inserted 10000000 Rows, about 142K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230706091915_clickbench_pr_173562.html

**Comment by eldenmoon** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.75 seconds
 stream load tsv:          498 seconds loaded 74807831229 Bytes, about 143 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          87.2 seconds inserted 10000000 Rows, about 114K ops/s
 storage size: 17162208579 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230706181850_clickbench_pr_173652.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.24 seconds
 stream load tsv:          449 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          68.8 seconds inserted 10000000 Rows, about 145K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230706103256_clickbench_pr_173650.html

**Comment by eldenmoon** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.76 seconds
 stream load tsv:          501 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          89.8 seconds inserted 10000000 Rows, about 111K ops/s
 storage size: 17164037682 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230708224246_clickbench_pr_174930.html

File: fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExec.java [hunk 1]
```diff
@@ -185,31 +185,36 @@ private RowBatch getNextInternal(Status status, Backend backend) throws TExcepti
...
                 long currentTs = System.currentTimeMillis();
                 if (currentTs >= timeoutTs) {
                     LOG.warn("fetch result timeout {}", backend.getBrpcAdress());
                     status.setStatus("query timeout");
                     return null;
                 }
                 try {
-                    pResult = futureResponse.get(timeoutTs - currentTs, TimeUnit.MILLISECONDS);
+                    pResult = futureResponse.get((timeoutTs - currentTs) / 3, TimeUnit.MILLISECONDS);
                 } catch (InterruptedException e) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExec.java:197]: why /3?

Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExec.java]: modified

**Comment by eldenmoon** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.67 seconds
 stream load tsv:          445 seconds loaded 74807831229 Bytes, about 160 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          68.3 seconds inserted 10000000 Rows, about 146K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230709120551_clickbench_pr_175209.html

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 52.71 seconds
 stream load tsv:          499 seconds loaded 74807831229 Bytes, about 142 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          86.8 seconds inserted 10000000 Rows, about 115K ops/s
 storage size: 17167058158 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230709201838_clickbench_pr_175207.html

File: fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExec.java [hunk 2]
```diff
@@ -201,15 +201,20 @@ private RowBatch getNextInternal(Status status, Backend backend) throws TExcepti
...
+                } catch (TimeoutException e) {
+                    futureResponse.cancel(true);
+                    LOG.warn("fetch result timeout {}, addr {}", timeoutTs - currentTs, backend.getBrpcAdress());
+                    status.setStatus("query timeout");
+                    return null;
                 }
             }
         } catch (RpcException e) {
-            LOG.warn("fetch result rpc exception {}", backend.getBrpcAdress());
+            LOG.warn("fetch result rpc exception {}, e {}", backend.getBrpcAdress(), e);
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExec.java:212]: should be the same as next LOG.warn

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**